File: Difficulty: Medium/Find length of Loop/find-length-of-loop.cpp
/*
class Node {
 public:
    int data;
    Node *next;

    Node(int x) {
        data = x;
        next = NULL;
    }
};
*/

class Solution {
  public:
    int lengthOfLoop(Node *head) {
        // If list is empty, no loop exists
        if(!head) return 0;
        
        Node* slow = head;
        Node* fast = head;
        
        // Phase 1: Detect if loop exists using Floyd's algorithm
        while(fast && fast->next) {
            slow = slow->next;          // Tortoise: moves 1 step
            fast = fast->next->next;    // Hare: moves 2 steps
            
            // If they meet, loop exists
            if(slow == fast) {
                // Phase 2: Measure loop length
                int length = 1;
                Node* temp = slow->next;
                
                // Move one pointer around the loop until it comes back
                while(temp != slow) {
                    length++;
                    temp = temp->next;
                }
                return length;  // Return the counted length
            }
        }
        
        return 0;  // No loop found
    }
};

File: Difficulty: Medium/Find length of Loop/find-length-of-loop.java
/*
class Node {
    int data;
    Node next;

    Node(int x) {
        data = x;
        next = null;
    }
}
*/

class Solution {
    public int lengthOfLoop(Node head) {
        // Here is the code 
        if(head == null || head.next == null){
            return 0;
        }
        int count = 0;
        Node slow = head;
        Node fast = head;
        while(fast != null && fast.next != null){
            slow=slow.next;
            fast = fast.next.next;
             if(slow == fast){
                 count = 1;
                 slow=slow.next;
                 while(fast!=slow){
                     slow=slow.next;
                     count++;
                 }
               break;
            }
        }
        
        return count;
    }
}